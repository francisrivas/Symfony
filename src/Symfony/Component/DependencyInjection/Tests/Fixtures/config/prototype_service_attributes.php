<?php

namespace Symfony\Component\DependencyInjection\Loader\Configurator;

use Symfony\Component\DependencyInjection\Tests\Fixtures\Prototype;

return function (ContainerConfigurator $c) {
    $di = $c->services()->defaults()
        ->tag('baz');

    $di->load(Prototype::class.'\\', '../Prototype')
        ->onlyWithServiceAttribute()
        ->public()
        ->autoconfigure()
        ->exclude('../Prototype/{BadClasses,BadAttributes}')
        ->factory('f')
        ->deprecate('vendor/package', '1.1', '%service_id%')
        ->args([0])
        ->args([1])
        ->tag('foo')
        ->parent('foo');

    $di->set('foo')->lazy()->abstract()->public();
};
