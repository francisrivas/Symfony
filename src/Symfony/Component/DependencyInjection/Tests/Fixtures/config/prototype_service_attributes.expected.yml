
services:
    service_container:
        class: Symfony\Component\DependencyInjection\ContainerInterface
        public: true
        synthetic: true
    Symfony\Component\DependencyInjection\Tests\Fixtures\Prototype\ServiceAttributes\BarWithService:
        class: Symfony\Component\DependencyInjection\Tests\Fixtures\Prototype\ServiceAttributes\BarWithService
        public: true
        tags:
            - foo
            - baz
        deprecated:
            package: vendor/package
            version: '1.1'
            message: '%service_id%'
        autoconfigure: true
        lazy: true
        arguments: [1]
        factory: f
    Symfony\Component\DependencyInjection\Tests\Fixtures\Prototype\ServiceAttributes\FooWithService:
        class: Symfony\Component\DependencyInjection\Tests\Fixtures\Prototype\ServiceAttributes\FooWithService
        public: true
        tags:
            - foo
            - baz
        deprecated:
            package: vendor/package
            version: '1.1'
            message: '%service_id%'
        autoconfigure: true
        lazy: true
        arguments: [1]
        factory: f
